git-pr-chain.py

```diff
@@ -750,6 +750,21 @@ def generate_pr_chain_annotation(commit_title) -> str:
     if not DRY_RUN:
         git("commit", "--amend", f"-m{new_commit_msg}")
 
+def cmd_end_chain(args):
+    # Use this command to create an empty commit that marks the end of the chain.
+    # Commits after this one will not be included in any PR when doing a push.
+    check_staged = git("diff", "--staged")
+
+    # Do not accidentally add changes to the chain end commit.
+    # It is easier for it to be empty and can be moved around in `git rebase`
+    if check_staged:
+        fatal("Please unstage all changes. Chain end commit should be empty")
+
+    chain_end_msg = "=== git-pr-chain: STOP ==="
+    print(f"Create empty commit to stop the chain: ", chain_end_msg)
+    if not DRY_RUN:
+        git("commit", "--allow-empty", "-m", chain_end_msg)
```

Comment:
For this command, since it's just `git commit --allow-empty -m "git pr-chain: STOP` -- do we really need a command in the tool for it?  Like, if you know how to use interactive rebase, which is kind of a prerequisite for using this at all, then should we assume you also know how to type that, or add an alias of your own?

I guess I lean towards having fewer commands so you don't have to learn / understand as much in order to use the tool.

Reply:
I think commands are abstractions... I want them to capture the intention of users, ie what they want to do, and simplify it for them. And this command capture that intention, and being convenient enough that you don't need to think about how it works behind the scene.

Reply:
I think `end-chain` is an important enough concept that the tool should support it formally. Also, by making it an abstraction (a command), we can hide or change the implementation however we want.

Reply:
I think you do need to think about how it works behind the scenes, though, because you're going to have to delete this commit to "un-stop" your chain.  You have to understand that deleting the commit is tantamount to un-stopping things, that there's no other metadata you need to delete.

Reply:
Yeah, I agree user still needs to understand the mechanism of "stopping the chain" (and we should document that to unstop the chain, remove the commit added.) On `git rebase -i` flow, it will look like this, and I think it's pretty clear what'll happen:
![Screen Shot 2020-10-21 at 11 08 09 AM](https://user-images.githubusercontent.com/687617/96760835-dc2a2c80-138d-11eb-81e8-2e263d7dcd3e.png)

Anyway, I'm an advocator for command-line aliases and ease of use, so I still find value in having this command. Even when it's something users can do manually, it's quite a lot to remember or to type to achieve something like this. Another benefit is that it'll show up on the command's help page. So when someone looks for a way to "stop adding commits to a PR", they can see it from the command line help and don't need to look for help from the repo's documentation.

Reply:
Okay, I guess that isn't so bad.

Apparently the behavior of git rebase has changed wrt empty commits?  Previously I remember it would drop them.  Now it seems that it will keep them around and annotate with `#empty`?  That's better for this use case!